### DISCUSSION TITLE
Error sk4d.dll

### DISCUSSION THREAD

**Discussion by kellyCardosoS** [skia4delphi/skia4delphi#246] [Q&A] [answered]
I'm using skia in a project, when it runs for the first time it doesn't give any type of error, but from the second time it runs it seems that the dll sk4d.dll is not in the project folder, but it is.
When I change the dll it works again, but in the second execution of the program the error again.
This only happens with this dll, you have already encountered this error, do you have any idea what it could be?

Using: Windows 10 - 64 bit
![image](https://github.com/skia4delphi/skia4delphi/assets/54945485/780b37f2-a5cd-4430-97cb-a410edf3b678)

**Comment by viniciusfbb** [comment]
Hello! Thank you very much for reporting the issue. This issue is definitely not normal and we need more information to understand the cause.

1. Do you have any other .exe or .dll in the same directory as your app, which uses Skia4Delphi? If so, please keep the same Skia4Delphi version for everyone.
2. Do you have more than 1 version of RAD Studio installed? If so, please install the same version of Skia4Delphi on all of them.
3. Are you doing some manual copying of the dll? If yes, this should not be done because the Skia4Delphi plugin already does this work during app compilation.
4. Is there any chance that you have some sk4d.dll in some directory of your PATHs? I do not recommend leaving the dll in a directory that is in the PATHs, ideally the dll must be together with the .exe to avoid causing problems in other applications that use Skia4Delphi.
5. Could you send the dll hash when the app gives an error and the dll hash when the app does not give an error? I suspect the dll is being mistakenly replaced for some reason...
6. Does the problem occur when you open the app directly in the explorer or when you run it from within the IDE?
7. What version of Skia4Delphi are you using? Did you do a manual install by any chance?

**Comment by kellyCardosoS** [comment]
Hello!
It's very strange I've been trying to solve it for a few days but no solution.

**1. Do you have any other .exe or .dll in the same directory as your app, which uses Skia4Delphi? If so, please keep the same Skia4Delphi version for everyone.**
There is no other exe or dll in the same directory.

**2. Do you have more than 1 version of RAD Studio installed? If so, please install the same version of Skia4Delphi on all of them.**
Yes, I have other versions of RAD Studio installed, but I uninstalled skia for all versions and reinstalled the same version.

**3. Are you doing some manual copying of the dll? If yes, this should not be done because the Skia4Delphi plugin already does this work during app compilation.**
I'm not doing manual copying of the dll.

**4. Is there any chance that you have some sk4d.dll in some directory of your PATHs? I do not recommend leaving the dll in a directory that is in the PATHs, ideally the dll must be together with the .exe to avoid causing problems in other applications that use Skia4Delphi.**
I don't have sk4d.dll in any PATHs directory, I'm leaving the DLL only in the exe directory.

**5. Could you send the dll hash when the app gives an error and the dll hash when the app does not give an error? I suspect the dll is being mistakenly replaced for some reason...**
When the error occurs: C6D0F964F77B228F0527E750F021B7D136E68DFE79862F4A4A3C4451BCCBC7FC
Working: 2539C9B5DE861C5C3927D966DCFFAC3BEE4BE0CF7B155A719AA91A82F2953E2B

**6.Does the problem occur when you open the app directly in the explorer or when you run it from within the IDE?**
The problem occurs when I run the app in explorer.

**What version of Skia4Delphi are you using? Did you do a manual install by any chance?**
I am using version v5.0.0, I performed the installation using Skia4Delphi_5.0.0_Setup.exe.

**Reply by viniciusfbb** [reply]
Clearly there is something wrong with the hash as it shouldn't have changed. What algorithm did you use to extract this hash? MD5?

**Reply by kellyCardosoS** [reply]
Yes, I used MD5 for extraction

**Reply by viniciusfbb** [reply]
I checked the MD5 of `skia4delphi\Binary\Shared\Win32\sk4d.dll` and `skia4delphi\Binary\Shared\Win64\sk4d.dll`, on version 5.0.0 and version 6.0.0-beta2. No official dll hash matched yours. I'm thinking of 2 possibilities:
1. Old version conflict
2. Virus (?)

You could try installing the latest version: [v6.0.0-beta2](https://github.com/skia4delphi/skia4delphi/releases), but select to install on all RAD Studios just to be safe...

**Reply by kellyCardosoS** [reply, marked as answer]
I couldn't identify the problem, but to solve it, I uninstalled all items related to RAD Studio and skia, and installed again using version 4.0.2, and I don't have the problem anymore, so far.

**Reply by viniciusfbb** [reply]
Right. I will close it now but few free to open a new discussion/issue if the problem return.
